jobs/integration/conftest.py

```diff
@@ -141,22 +153,52 @@ async def run(self, cmd, *args, stdin=None):
         if hasattr(stdin, "encode"):
             stdin = stdin.encode("utf8")
 
-        stdout, stderr = await proc.communicate(input=stdin)
-        if proc.returncode != 0:
+        stdout, stderr = bytearray(), bytearray()
+
+        def tee(line: bytes, sink: bytearray, fd: int):
+            sink += line
+            write = _tee == "out" and fd == 1
+            write |= _tee == "err" and fd == 2
+            if write or _tee is True:
+                os.write(fd, line)
```

Comment:
append the subprocess `stdout` and `stderr` to a `bytesarray` object while writing to the test's `stdout` and `stderr`